### ISSUE TITLE
Custom global widget not rendering correct content in page preview view

### ISSUE THREAD

**Issue by LeonSkrilec** [elementor/elementor#12631]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

I have a global custom widget inserted in the page. The widget is rendering correctly on the front-end. However if I open the page in the Elementor editor mode the widget is not displayed correctly, as it does not show user entered content. If I click on the widget, select Edit global widget, make a slight change to widgets title, correct content appears.

It looks like Elementor is not fetching correct global widget content on initial page preview load. 

*Issue only exists for custom widgets. Default Elementor widgets work fine.*

## Steps to reproduce

1. Insert a custom widget in your site and fill it's fields with content
2. Make custom widget global
3. Insert global widget on other site
4. Content inserted in 1. is not showing

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MariaDB Server v10.3.24
	PHP Version: 7.3.22
	PHP Max Input Vars: 10000
	PHP Max Post Size: 512M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.5.1
	Site URL: https://kranjska-gora.forwardapps.si
	Home URL: https://kranjska-gora.forwardapps.si
	WP Multisite: No
	Max Upload Size: 512 MB
	Memory limit: 1024M
	Permalink Structure: /%postname%/
	Language: sl-SI
	Timezone: Europe/Ljubljana
	Debug Mode: Inactive

== Theme ==
	Name: Kranjska Gora
	Version: 1
	Author: Leon Škrilec
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: sl_SI
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields Multilingual
		Version: 1.7.3
		Author: OnTheGoSystems

	Advanced Custom Fields PRO
		Version: 5.8.12
		Author: Elliot Condon

	Akismet Anti-Spam
		Version: 4.1.6
		Author: Automattic

	Classic Editor
		Version: 1.6
		Author: WordPress Contributors

	Custom Post Type UI
		Version: 1.8.0
		Author: WebDevStudios

	DynamicConditions
		Version: 1.4.6
		Author: RTO GmbH

	Elementor
		Version: 3.0.9
		Author: Elementor.com

	Elementor Pro
		Version: 3.0.4
		Author: Elementor.com

	Intuitive Custom Post Order
		Version: 3.1.2
		Author: hijiri

	jQuery Manager for WordPress
		Version: 1.10.6
		Author: Remzi Cavdar

	Kranjska Gora Elementor widgets
		Version: 1.0
		Author: Leon Škrilec

	Post Type Switcher
		Version: 3.2.0
		Author: John James Jacoby

	Redirection
		Version: 4.8
		Author: John Godley

	Regenerate Thumbnails
		Version: 3.1.3
		Author: Alex Mills (Viper007Bond)

	TinyMCE Advanced
		Version: 5.5.0
		Author: Andrew Ozz

	WPML Media
		Version: 2.6.0
		Author: OnTheGoSystems

	WPML Multilingual CMS
		Version: 4.4.2
		Author: OnTheGoSystems

	WPML String Translation
		Version: 3.1.2
		Author: OnTheGoSystems

	WPML Translation Management
		Version: 2.10.1
		Author: OnTheGoSystems

	Yoast Duplicate Post
		Version: 3.2.6
		Author: Enrico Battocchi & Team Yoast

	Yoast SEO
		Version: 14.9
		Author: Team Yoast

	Yoast SEO Multilingual
		Version: 1.2.4
		Author: OnTheGoSystems


== Must-Use Plugins ==
	Elementor Safe Mode
		Version: 1.0.0
		Author: Elementor.com


== Elements Usage ==
	
	archive : 44
		DynamicGrid : 9
		PageHero : 6
		PostsGrid : 1
		PostsSlider : 1
		accomodationCategories : 2
		global : 24
	single : 36
		PostFields : 4
		PostHero : 6
		PostsSlider : 5
		PriceAndBook : 2
		backdroptitle : 5
		gallery : 5
		global : 11
		post-comments : 1
		post-navigation : 1
		theme-post-content : 9
		theme-post-excerpt : 1
		theme-post-title : 3
	widget : 0
		ContactBoxes : 68
		CubilisBooking : 6
		PageHero : 33
		PostsSlider : 173
		StoriesGrid : 26
		accomodationCategories : 115
	wp-post : 93
		ContactBoxes : 3
		PostsSlider : 1
		TrailForks : 1
		accordion : 1
		backdroptitle : 1
		button : 11
		call-to-action : 4
		divider : 15
		gallery : 13
		global : 2
		google_maps : 9
		heading : 10
		html : 10
		image : 124
		popularPosts : 1
		portfolio : 1
		social-icons : 1
		spacer : 13
		tabs : 1
		text-editor : 345
		toggle : 1
		video : 28
	wp-page : 45
		PageHero : 2
		PostsSlider : 7
		Social : 4
		Weather : 1
		Webcams : 1
		accordion : 5
		backdroptitle : 4
		global : 27
		heading : 18
		hero : 4
		image : 42
		popularPosts : 4
		spacer : 6
		tabs : 1
		text-editor : 13
		video : 3


== Log ==
	
JS: showing 12 of 12JS: 2020-08-03 15:03:29 [error X 2][https://kranjska-gora.test/wp-admin/load-scripts.php?c=1&loadchunk_0=underscore,jquery-core,backbone,jquery-ui-core,jquery-ui-widget,jquery-ui-mouse,shortcode,utils,wp-util,wp-backbone,media-models&loadchunk_1=,moxiejs,plupload,wp-plupload,jquery-ui-sortable&ver=5.4.2:2:15088] Unexpected token \')\' 
JS: 2020-08-05 06:46:05 [error X 1][https://kranjska-gora.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view1851\") has already been destroyed and cannot be used. 
JS: 2020-08-08 10:04:34 [error X 3][https://kranjska-gora.test/wp-content/plugins/elementor/assets/js/editor-document.min.js?ver=2.9.13:2:195519] Cannot read property \'map\' of undefined 
JS: 2020-08-11 09:55:30 [error X 1][https://kranjska-gora.test/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=2.9.13:2:59543] Cannot read property \'attributes\' of undefined 
JS: 2020-08-11 13:13:04 [error X 24][https://kranjska-gora.test/wp-content/plugins/elementor-pro/assets/js/frontend.min.js?ver=2.10.3:2:58831] Cannot read property \'setSettings\' of undefined 
JS: 2020-08-11 14:10:46 [error X 6][https://kranjska-gora.test/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.9.13:31:118543] this.$el.find(...).tipsy is not a function 
JS: 2020-08-15 12:39:52 [error X 1][https://kranjska-gora.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view8912\") has already been destroyed and cannot be used. 
JS: 2020-09-02 19:04:10 [error X 1][https://kranjska-gora.forwardapps.si/wp-content/plugins/jquery-manager/assets/js/jquery-3.5.1.min.js:2:85980] r.getClientRects is not a function 
JS: 2020-09-10 07:25:48 [error X 1][https://kranjska-gora.forwardapps.si/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view308\") has already been destroyed and cannot be used. 
JS: 2020-09-10 07:26:44 [error X 1][https://kranjska-gora.forwardapps.si/wp-includes/js/jquery/jquery.js?ver=1.12.4-wp:2:1825] cannot call methods on resizable prior to initialization; attempted to call method \'destroy\' 
JS: 2020-09-18 16:29:22 [error X 2][https://kranjska-gora.forwardapps.si/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.0.8.1:2:473779] Cannot read property \'document\' of undefined 
JS: 2020-09-23 07:20:13 [error X 1][https://kranjska-gora.forwardapps.si/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view1057\") has already been destroyed and cannot be used. 

PHP: showing 7 of 7PHP: 2020-08-11 15:12:32 [notice X 35][/Applications/MAMP/htdocs/kranjska-gora/wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-gallery.php::37] Undefined offset: 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-08-11 15:12:35 [warning X 6][/Applications/MAMP/htdocs/kranjska-gora/wp-content/plugins/elementor/core/common/modules/ajax/module.php::269] Cannot modify header information - headers already sent by (output started at /Applications/MAMP/htdocs/kranjska-gora/wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-gallery.php:37) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-08-13 08:40:08 [error X 1][/Applications/MAMP/htdocs/kranjska-gora/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::259] Uncaught Error: Call to a member function update_meta() on boolean in /Applications/MAMP/htdocs/kranjska-gora/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:259
Stack trace:
#0 /Applications/MAMP/htdocs/kranjska-gora/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(180): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(NULL, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /Applications/MAMP/htdocs/kranjska-gora/wp-content/plugins/elementor/core/common/modules/ajax/module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 /Applications/MAMP/htdocs/kranjska-gora/wp-includes/class-wp-hook.php(287): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 /Applications/MAMP/htdocs/kranjska-gora/wp-includes/class-wp-h [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-08-18 09:26:54 [warning X 18][/home/fwdapp/kranjska-gora.forwardapps.si/wp-content/plugins/elementor/core/common/modules/ajax/module.php::269] Cannot modify header information - headers already sent by (output started at /home/fwdapp/kranjska-gora.forwardapps.si/wp-content/plugins/sitepress-multilingual-cms/sitepress.class.php:4214) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-08-24 13:00:13 [notice X 7][/home/fwdapp/kranjska-gora.forwardapps.si/wp-content/plugins/elementor-pro/modules/posts/skins/skin-content-base.php::248] Trying to get property 'ID' of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-08-24 14:25:27 [notice X 1][/home/fwdapp/kranjska-gora.forwardapps.si/wp-content/plugins/elementor/includes/managers/image.php::88] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-08-25 12:08:11 [notice X 4][/home/fwdapp/kranjska-gora.forwardapps.si/wp-content/plugins/elementor/includes/embed.php::156] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Log: showing 20 of 242020-08-27 13:11:08 [info] Elementor/Upgrades - _v_3_0_0_move_general_settings_to_kit Finished 
2020-08-27 13:11:08 [info] Elementor/Upgrades - _v_3_0_0_move_default_colors_to_kit Start  
2020-08-27 13:11:08 [notice] System colors are disabled. nothing to do. 
2020-08-27 13:11:08 [info] Elementor/Upgrades - _v_3_0_0_move_default_colors_to_kit Finished 
2020-08-27 13:11:08 [info] Elementor/Upgrades - _v_3_0_0_move_saved_colors_to_kit Start  
2020-08-27 13:11:08 [info] Elementor/Upgrades - _v_3_0_0_move_saved_colors_to_kit Finished 
2020-08-27 13:11:08 [info] Elementor/Upgrades - _v_3_0_0_move_default_typography_to_kit Start  
2020-08-27 13:11:08 [notice] System typography is disabled. nothing to do. 
2020-08-27 13:11:08 [info] Elementor/Upgrades - _v_3_0_0_move_default_typography_to_kit Finished 
2020-08-27 13:11:08 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.9.14',
  'to' => '3.0.2',
)]
2020-09-01 10:47:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.10.3',
  'to' => '3.0.2',
)]
2020-09-01 10:47:28 [info] elementor::elementor_updater Started 
2020-09-01 10:47:28 [info] Elementor/Upgrades - _v_3_0_5_re_move_space_between_widgets_to_kit Start  
2020-09-01 10:47:28 [info] Elementor/Upgrades - _v_3_0_5_re_move_space_between_widgets_to_kit Finished 
2020-09-01 10:47:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.2',
  'to' => '3.0.5',
)]
2020-09-01 10:47:28 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.2',
  'to' => '3.0.5',
)]
2020-09-01 10:47:29 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.10.3',
  'to' => '3.0.2',
)]
2020-09-15 12:35:53 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.5',
  'to' => '3.0.8.1',
)]
2020-09-17 15:53:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.2',
  'to' => '3.0.4',
)]
2020-09-23 09:40:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.8.1',
  'to' => '3.0.9',
)]


				
```
</details>

**Comment by MindsEyeDesign** [comment]
I have the same issue with global widgets appearing as grey boxes in admin. They're not hidden in responsive settings of pulling in dynamic content, just a divider and a small gallery. I've sent a ticket to support with a demo site set up for them to test. Just waiting to hear back.

**Comment by saulposel** [comment]
Just confirming that I have the same issue with the latest Elementor / Pro versions.

**Comment by shilo-ey** [comment]
Hi @saulposel  @MindsEyeDesign @LeonSkrilec 

Do you still experience this issue with the latest versions of Elementor? it was supposed to be fixed.

Thanks

**Comment by LeonSkrilec** [comment]
Hey @shilo-ey ,

I am on Elementor v 3.0.14 and Elementor PRO v 3.0.8 and issue is still present.
I tried removing and creating new widget and making it global.
I attached .mp4 screen recording confirming the issue.

https://user-images.githubusercontent.com/6199422/102768359-54f33800-4381-11eb-93d6-8f3285f7bb4c.mp4

**Comment by Aviad-Herman** [comment]
Hello,
Is this still an issue or has it been solved in the latest versions of Elementor?

If the bug persists, we recently updated the infrastructure behind Global Widgets. The update is currently only available on Elementor's developer edition releases (https://elementor.com/blog/introducing-beta-developer-edition/). Could you see if the bug still happens on the developer release?

Important note: the developer releases should not be used on live sites

**Comment by MindsEyeDesign** [comment]
I'm having terrible issues with the latest Elementor Pro and Global Widgets (GW). 

I'm still getting grey boxes designating where the GW is but also widgets are now corrupting back to the widget's default setup even though they had been styled and saved as GWs. 

When trying to edit a widget, sometimes nothing shows up in the left edit column!

If I click on a placed Global Widget and try to move it, I sometimes get a repeating blue line/box down the column it's in and a box saying what the widget is, appears in the top left of the column, which I cannot do anything with.

Sometimes, if I right-click on a GW and go to delete it, it deletes the whole section.

Basically, GWs are a real mess at the moment and really frustrating me and my clients.

I do not see why I should spin up a copy with the latest dev release either to let you know if it works. It would be much better if the Elementor team created a stable and bulletproof product that was rock solid rather than continuously releasing updates and new features. Building it a little slower and getting updates right would be far more helpful.

**Comment by Aviad-Herman** [comment]
I'm sorry to hear that and totally agree.

Our QA process has/is undergoing great changes. More than 80% of the product is validated by automated visual regression tests in addition to manual QA. The problem is the various parts of the ecosystem - themes, plugins, servers, php versions, browsers etc. and the problem is, being open source means everyone can make changes to our code.

Most SAAS companies have the option to run a gradual release process and catch bugs before they are released to the general public. Unfortunately, WordPress guidelines prohibit us from doing so. All we can rely on is the developer edition and beta releases.

We are continually working to cover more configurations, hope you will see all these problems fixed in Pro 3.5.0.

**Comment by blaifm** [comment]
Can confirm this issue is still not resolved in dev release 3.5.0. My custom global widget (actually an _Essential Addons for Elementor_ widget) appears as a grey box in the editor and loses format in frontend when used as a global widget.
